src/runtime/runtime.cpp

```diff
@@ -272,12 +197,51 @@ main(int argc, char* argv[]) {
 
     std::unique_ptr<context_t> context;
 
+    signal::handler_t signal_set;
+    signal_set.set_default_action(SIGPIPE, signal::handler_base_t::default_action_t::IGNORE);
+
+    std::function<void(const std::error_code& ec, int)> cb = [&](const std::error_code& ec, int) {
+        if(ec) {
+            return;
+        }
+        COCAINE_LOG_INFO(logger, "resetting logger");
+        logging.reset(*logger, backend);
+        signal_set.async_wait_simple(SIGHUP, cb);
+    };
+    signal_set.async_wait_simple(SIGHUP, cb);
+
+    std::thread sig_thread([&](){
```

Comment:
thread dedicated for running signal handler. In general we want running signal handler during context_t destruction to handle work such as collecting childs. So we can not block for run in main thread.